include/ozo/impl/connection.h

```diff
@@ -95,10 +94,8 @@ void connection<OidMap, Statistics>::async_wait_read(WaitHandler&& h) {
 
 template <typename OidMap, typename Statistics>
 error_code connection<OidMap, Statistics>::close() noexcept {
-    error_code ec;
-    socket_.close(ec);
```

Comment:
Nope, since the socket is managed within PGconn object and will be closed right at the object destruction. Previously the socket_ object had a copy of connection socket file descriptor that obtained via dup() function so the close() call was necessary. Now there is no dup() call, so the socket_ object should not manage the descriptor. But reset() will cancel all the IO on the socket_.